test_programs/execution_success/cast_to_i8_regression_7776/Prover.toml

```diff
@@ -0,0 +1,3 @@
+a = 0
+b = 1
+return = 0
```

Comment:
Sorry to ask basic questions, but why does `-1` as `Field` turn into `0` as `i8`?

Reply:
This program succeeds:
```
pub fn main() {
    let x: Field = -2;
    let z: i8 = -1;
    assert_eq(x as i8, z)
}
```

So does this:
```rust
pub fn main() {
    let x: Field = 0;
    let y: Field = -1;
    let z: i8 = 0;
    assert_eq(x as i8, z);
    assert_eq(y as i8, z);
}
```

Is it just me who finds this unintuitive?

Reply:
Now, as for your other question of "is this unintuitive?"

Yes. :slightly_smiling_face: